app/ui/src/app/import-export/import/import.actions.ts

```diff
@@ -0,0 +1,117 @@
+import { Action } from '@ngrx/store';
+
+import { ActionReducerError } from '@syndesis/ui/platform';
+import {
+  IntegrationImportRequest,
+  IntegrationUploadRequest,
+} from './import.models';
+import { Integrations} from '@syndesis/ui/model';
+
+export class IntegrationImportActions {
+  static FETCH = '[Import Integration] Fetch integrations request';
+  static FETCH_COMPLETE = '[Import Integration] Fetch operation complete';
+  static FETCH_FAIL = '[Import Integration] Fetch operation failed';
+  static IMPORT = '[Import Integration] Import integration(s) request';
+  static IMPORT_COMPLETE = '[Import Integration] Import integration complete';
+  static IMPORT_FAIL = '[Import Integration] Import integration failed';
+  static UPLOAD = '[Import Integration] Upload integration for import request';
+  static UPLOAD_CANCEL = '[Import Integration] Upload integration for import cancelled';
+  static UPLOAD_COMPLETE = '[Import Integration] Upload integration for import complete';
+  static UPLOAD_FAIL = '[Import Integration] Upload integration for import failed';
+
+
+  /**
+   * Not actually sure that we would need fetching here.
+   */
+  static fetch() {
+    return new IntegrationFetch();
+  }
+
+  static fetchComplete(payload: Integrations): IntegrationFetchComplete {
+    return new IntegrationFetchComplete(payload);
+  }
+
+  static import(payload: IntegrationImportRequest): IntegrationImport {
+    return new IntegrationImport(payload);
+  }
+
+  static importFail(payload: ActionReducerError): IntegrationImportFail {
+    return new IntegrationImportFail(payload);
+  }
+
+  static importComplete(payload: any): IntegrationImportComplete {
+    return new IntegrationImportComplete(payload);
+  }
+
+  static upload(payload: IntegrationUploadRequest): IntegrationUpload {
+    return new IntegrationUpload(payload);
+  }
+
+  static uploadFail(payload: ActionReducerError): IntegrationUploadFail {
+    return new IntegrationUploadFail(payload);
+  }
+
+  static uploadComplete(payload: any): IntegrationUploadComplete {
+    return new IntegrationUploadComplete(payload);
+  }
+
+  static uploadCancel(): IntegrationUploadCancel {
+    return new IntegrationUploadCancel();
+  }
+}
```

Comment:
The idea for creating static methods in a Actions class to dispatch actions more easily eventually proved to generate too much boilerplate. Perhaps is better to use Action classes as is. Again, you can see the leaner approach I took for the metrics domain asa reference.